flow/core/kernel/toyota_energy.py

```diff
@@ -25,18 +25,19 @@ def __init__(self, kernel, filename=None):
         os.remove(file.pkl)
     
     @abstractmethod
-    def get_instantaneous_power(self):
+    def get_instantaneous_power(self, accel, speed, grade):
         pass
 
 
 class PriusEnergy(ToyotaModel):
     
     def __init__(self, kernel):
         super(PriusEnergy, self).__init__(kernel, filename = 'prius_test.pkl')
+        self.soc = # how to pass here the parameter vargument from the rewards.py

```

Comment:
This is where i should use the value passed in the "parameter" in the function in rewards.py.

Reply:
so i think this is where you use the `args` or `kwargs` in the `__init__` call signature to get `initial_soc`.

you'll just initialize it in VehicleParams (with a default value of 0.9, for example). you won't need to pass a new `soc` every time from rewards.py because this class should be tracking/updating the soc at every time_step